server/paymentProviders/stripe/creditcard.js

```diff
@@ -93,37 +93,37 @@ const createChargeAndTransactions = async (hostStripeAccount, { order, hostStrip
   // Make sure data is available (breaking in some old tests)
   order.data = order.data || {};
 
-  let paymentIntent;
-  if (!order.data || !order.data.paymentIntent) {
-    const payload = {
+  let paymentIntent = order.data && order.data.paymentIntent;
```

Comment:
Looking at the line above (`order.data = order.data || {};`) I think we can do directly 

```suggestion
  let paymentIntent = order.data.paymentIntent;
```